package.cmake

```diff
@@ -21,6 +21,20 @@ if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.9" AND EXISTS /etc/os-release)
       string(APPEND CPACK_PACKAGE_FILE_NAME "-el${CMAKE_MATCH_1}")
    endif()
 endif()
+
+# Fix debian package filename as it should have the format:
+# <PackageName>_<VersionNumber>_<DebianArchitecture>.deb
+
+# Find architecture using dpkg
```

Comment:
I believe this is already available via CMake's `CPACK_DEBIAN_PACKAGE_ARCHITECTURE` variable. Can you try using that instead?

Reply:
The problem with `CPACK_DEBIAN_PACKAGE_ARCHITECTURE` is that it should be set in the script. cpack will use `dpkg --print-architecture` only if its empty, after it has parsed `package.cmake`

We also set `CPACK_DEBIAN_<COMPONENT>_FILE_NAME` filename ourself. so cpack has no way to inject it's default architecture value into the filename.

i could however set `CPACK_DEBIAN_PACKAGE_ARCHITECTURE` instead of `DEB_ARCH`.